### PR TITLE
Favorite Emotes

### PR DESCRIPTION
**Pull request by Mitchdev**
https://github.com/destinygg/chat-gui/assets/13901269/0ea78878-4d26-455b-85dc-397b4ada92f4

### CODE REVIEW

File: assets/chat/css/style.scss [hunk 1]
```diff
@@ -1317,6 +1317,16 @@ hr {
     background: none;
     border-radius: 0;
   }
+  .all {
+    flex: 1;
+    height: auto;
```
Comment (11k) [inline assets/chat/css/style.scss:1322]: Elements have a height of `auto` by default. There's no need for this rule.

File: assets/chat/css/style.scss [hunk 2]
```diff
@@ -1317,6 +1317,16 @@ hr {
...
+  }
+  .favorite {
+    flex: 0;
+    height: auto;
+    min-height: 88px;
```
Comment (11k) [inline assets/chat/css/style.scss:1327]: Is the goal here to make the favorite panel a fixed 88px and have the all panel take up the remaining space? If so, you can simply give the favorite panel
```
flex: 0 88px;
```
which sets its initial height to 88px and tells it not to grow. With this rule alone, I believe both panels will behave as intended. You can remove everything else.

File: assets/chat/css/style.scss [hunk 3]
```diff
@@ -1491,6 +1502,23 @@ hr {
       &.creator {
         color: $text-color1;
       }
+      &.favorite {
+        cursor: pointer;
```
Comment (11k) [inline assets/chat/css/style.scss:1506]: If you make the element a button, which is more semantically correct anyway, you won't have to set the cursor type. You might have to remove the default button styles if there isn't a reset somewhere in this file already, though.

File: assets/chat/css/style.scss [hunk 4]
```diff
@@ -1491,6 +1502,23 @@ hr {
...
+        margin-top: 5px;
+        .btn-icon {
+          transition: background-color 500ms;
+          width: 20px;
+          height: 20px;
```
Comment (11k) [inline assets/chat/css/style.scss:1511]: With a square element it's better to specify the size in a single dimension and an aspect ratio of 1.

File: assets/chat/css/style.scss [hunk 5]
```diff
@@ -1491,6 +1502,23 @@ hr {
...
+          mask: url('../img/icon-favorite.svg') no-repeat;
+          background-color: $text-color2;
+        }
+        &.favorited .btn-icon {
+          mask: url('../img/icon-favorite-fill.svg') no-repeat;
```
Comment (11k) [inline assets/chat/css/style.scss:1516]: I got CORS errors trying to load this resource. Would you mind double-checking the NGINX config to see if any changes have to be made to allow loading SVGs from the CDN?

**Review by 11k — CHANGES_REQUESTED** [review]

**Comment by Mitchdev** [comment]
Updated NGINX @11k 
https://github.com/destinygg/website-private/pull/271
https://github.com/11k/dockerstiny/pull/24

File: assets/chat/css/style.scss (hunk 3 above)
Comment (Mitchdev) [reply assets/chat/css/style.scss:1506]: The button element has a default cursor so you'd still need to set it to cursor. I can still change it to a button element if you want.

Comment (11k) [reply assets/chat/css/style.scss:1500]: You know what, I'm so used to CSS frameworks setting the default cursor for buttons to pointer I assumed that was the browser default. That's pretty funny.

Sure, you can go ahead and make it a button. You can also define reset styles for buttons at the top of the file that gets rid of its background, border, and sets the cursor type, too. I believe the only other place buttons are used are in the overlay that pops up if you try to send a message while logged out. Digging through our HTML, it's alarming how many times we use anything but a button when a button is most appropriate. Something to address in the future.

**Review by 11k — APPROVED** [review]

**Pull request merged**